ee/app/livechat-enterprise/client/views/livechatUnitForm.js

```diff
@@ -0,0 +1,234 @@
+	},
+	departments() {
+		return Template.instance().unit && !_.isEmpty(Template.instance().unit.get()) ? Template.instance().unit.get().departments : [];
+	},
+	selectedMonitors() {
+		return _.sortBy(Template.instance().selectedMonitors.get(), 'username');
+	},
+	monitorsBeingSelected() {
+		return _.sortBy(Template.instance().monitorsBeingSelected.get(), 'username');
+	},
+	selectedDepartments() {
+		return _.sortBy(Template.instance().selectedDepartments.get(), 'name');
+	},
+	availableDepartments,
+	hasAvailableDepartments() {
+		return availableDepartments().length > 0;
+	},
+	monitorAutocompleteExceptions() {
+		return _.pluck(Template.instance().selectedMonitors.get(), 'username').concat(_.pluck(Template.instance().monitorsBeingSelected.get(), 'username'));
+	},
+	monitorModifier() {
+		return (filter, text = '') => {
+			const f = filter.get();
+			return `@${
+				f.length === 0
+					? text
+					: text.replace(
+						new RegExp(filter.get()),
+						(part) => `<strong>${ part }</strong>`,
+					)
+			}`;
+		};
+	},
+	onSelectMonitors() {
+		return Template.instance().onSelectMonitors;
+	},
+	availableMonitors() {
+		return Template.instance().availableMonitors.get();
+	},
+	onClickTagMonitors() {
+		return Template.instance().onClickTagMonitors;
+	},
+});
+
+Template.livechatUnitForm.events({
+	'submit #unit-form'(e, instance) {
+		e.preventDefault();
+		const $btn = instance.$('button.save');
+
+		const _id = $(e.currentTarget).data('id');
+		const visibility = instance.$('input[name=visibility]:checked').val();
+		const name = instance.$('input[name=name]').val();
+
+		if (visibility !== 'private' && visibility !== 'public') {
+			return toastr.error(t('Please_select_visibility'));
+		}
+
+		if (name.trim() === '') {
+			return toastr.error(t('Please_fill_a_name'));
+		}
+
+		const oldBtnValue = $btn.html();
+		$btn.html(t('Saving'));
+
+		const unitData = {
+			visibility,
+			name: name.trim(),
+		};
+
+		const unitMonitors = [];
+
+		instance.selectedMonitors.get().forEach((monitor) => {
+			unitMonitors.push(monitor);
+		});
+
+		const unitDepartments = [];
+		instance.selectedDepartments.get().forEach((department) => {
+			unitDepartments.push(department);
+		});
+
+		Meteor.call('livechat:saveUnit', _id, unitData, unitMonitors, unitDepartments, function(error/* , result*/) {
+			$btn.html(oldBtnValue);
+			if (error) {
+				return handleError(error);
+			}
+
+			toastr.success(t('Saved'));
+			FlowRouter.go('livechat-units');
+		});
+	},
+
+	async 'click .add-monitor'(e, instance) {
+		e.preventDefault();
+
+		const monitors = instance.monitorsBeingSelected.curValue;
+		if (!monitors.length) {
+			return toastr.error(t('Please_fill_a_username'));
+		}
+
+		const newMonitorList = [];
+		const selectedMonitors = instance.selectedMonitors.get();
+
+		for (const monitor of monitors) {
+			const monitorData = await APIClient.v1.get(`livechat/monitors.getOne?username=${monitor.username}`); //eslint-disable-line
```

Comment:
This call needs to be wrapped in a `try / catch` statement, otherwise, nothing will happen on the client in case the server throws an error like displayed below:

![Screen Shot 2020-03-18 at 11 13 34](https://user-images.githubusercontent.com/59577424/76969877-c50ff180-6909-11ea-922a-8388b4b25d47.png)